### PR TITLE
chore(assert): add experiment to assert if Recipe labels are set

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

## Pull Request template
Please, go through these steps before you submit a PR.

1. This repository follows semantic versioning convention, therefore each PR title/commit message must follow convention: `<type>(<scope>): [#issue_number] - <subject>`. 
   `type` is defining if release will be triggering after merging submitted changes, details in [CONTRIBUTING.md](../CONTRIBUTING.md). `#issue_number` is optional, when commit resolves any github issue.
    Most common types are:
    * `feat` - for new features
    * `fix` - for bug fixes or improvements
    * `chore` - changes not related to code

IMPORTANT: Please review the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

**PLEASE REMOVE THIS TEMPLATE BEFORE SUBMITTING**

### CODE REVIEW

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+apiVersion: dope.mayadata.io/v1
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:1]: rename the file to `assert-recipe-labels-and-phase-are-set`

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+kind: Recipe
+metadata:
+  name: assert-absence-of-deprecated-daemonsets
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:4]: rename this

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  namespace: d-testing
+  labels:
+    d-testing.dope.mayadata.io/inference: "true"
+spec:
+  tasks:
+  - name: assert-daemonset-with-extensions-v1beta1
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:10]: rename appropriately

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+apiVersion: dope.mayadata.io/v1
+kind: Recipe
+metadata:
+  name: assert-absence-of-deprecated-daemonsets
+  namespace: d-testing
+  labels:
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:6]: remove this labels

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    d-testing.dope.mayadata.io/inference: "true"
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:7]: remove this label, since inference targets the primary recipes & not the helper recipes.

Comment (AmitKumarDas) [reply test/experiments/assert-labels-and-phase-are-set.yaml:7]: add some specific labels e.g. 
```yaml
labels:
  assert-phase-is-set-in-label: true
```

File: test/experiments/assert-labels-and-phase-are-set.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  name: assert-labels-set-after-reconciliation
+  namespace: d-testing
+  labels:
+    d-testing.dope.mayadata.io/inference: "true"
+spec:
+  eligible:
+    checks:
+    - labelSelector:
+        matchExpressions:
+        - key: recipe.dope.mayadata.io/phase
```
Comment (AmitKumarDas) [inline test/experiments/assert-labels-and-phase-are-set.yaml:34]: make the selection specific to the helper recipe

**Comment by niladrih** [comment]
Addressing the changes in a separate PR.

**Pull request closed without merging**